httpx/_client.py

```diff
@@ -1196,6 +1201,11 @@ def __exit__(
         exc_value: BaseException = None,
         traceback: TracebackType = None,
     ) -> None:
+        if self._state != ClientState.OPENED:
+            raise RuntimeError(
+                "Cannot close a client, as the current state is not 'OPENED'."
```

Comment:
I'm not sure this is the most obvious error message either. Might be helpful to reference what the state is so that if it's already closed that would be obvious in the message.